cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -15,46 +15,368 @@
  */
 #pragma once
 
+#include <prims/extract_transform_v_frontier_outgoing_e.cuh>
+#include <prims/per_v_transform_reduce_incoming_outgoing_e.cuh>
+#include <prims/transform_reduce_v_frontier_outgoing_e_by_dst.cuh>
+#include <prims/update_edge_src_dst_property.cuh>
+#include <prims/update_v_frontier.cuh>
+#include <prims/vertex_frontier.cuh>
+
 #include <cugraph/algorithms.hpp>
+#include <cugraph/detail/utility_wrappers.hpp>
+#include <cugraph/edge_src_dst_property.hpp>
 #include <cugraph/utilities/error.hpp>
+#include <cugraph/vertex_partition_device_view.cuh>
+
+#include <thrust/functional.h>
+#include <thrust/reduce.h>
 
 #include <raft/core/handle.hpp>
 
+//
+// The formula for BC(v) is the sum over all (s,t) where s != v != t of
+// sigma_st(v) / sigma_st.  Sigma_st(v) is the number of shortest paths
+// that pass through vertex v, whereas sigma_st is the total number of shortest
+// paths.
+namespace {
+
+template <typename vertex_t>
+struct brandes_e_op_t {
+  const vertex_t max_vertex_t{std::numeric_limits<vertex_t>::max()};
+
+  template <typename value_t, typename ignore_t>
+  __device__ thrust::optional<value_t> operator()(
+    vertex_t, vertex_t, value_t src_property, vertex_t dst_property, ignore_t) const
+  {
+    thrust::optional<value_t> result{thrust::nullopt};
+    return (dst_property < max_vertex_t) ? result : thrust::make_optional(src_property);
```

Comment:
I guess `return (dst_distance == invalid_distance)? thrust::make_optional(src_sigma) : thrust::nullopt` will be more readable.